pkg/cmd/operator/webhooks_test.go

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) 2021 ScyllaDB
+
+package operator
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"net"
+	"net/http"
+	"os"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/scylladb/scylla-operator/pkg/genericclioptions"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/server/dynamiccertificates"
+)
+
+func TestComplete(t *testing.T) {
+	now := time.Now()
+	_, cert, privateKey, _, _, err := createCertificates(now)
+	if err != nil {
+		panic(err)
+	}
+
+	ts := []struct {
+		Name           string
+		WebhookOptions *WebhookOptions
+		ExpectedError  error
+		Callback       func(*WebhookOptions, *testing.T)
```

Comment:
could this use `ExpectedWebhookOptions` pattern?

Reply:
`Certificate's serial number is generated randomly so that's an issue. DynamicCertKeyPairContent would require a custom comparator and a way to avoid concurrent reads. Additionally `GetCertificate` can't be compared at all.

Reply:
maybe we can split it into a table based test for those that fit and one non-table based for the live reload. On the other hand that case could be cover through the Run flow which should call Complete anyways I think. In the end we primarily care about the reload test, the rest is covered through other flows in our suite.